packages/core/src/useForm/useFormActions.ts

```diff
@@ -170,31 +173,86 @@ export function useFormActions<TForm extends FormObject = FormObject, TOutput ex
     }
   }
 
-  async function reset(state?: Partial<ResetState<TForm>>, opts?: SetValueOptions) {
-    if (state?.values) {
-      form.setInitialValues(state.values, opts);
+  async function reset(): Promise<void>;
+  async function reset(state: Partial<ResetState<TForm>>, opts?: SetValueOptions): Promise<void>;
+  async function reset<TPath extends Path<TForm>>(
+    path: TPath,
+    state: Partial<ResetState<TForm>>,
+    opts?: SetValueOptions,
+  ): Promise<void>;
+
+  // Implementation signature
+  async function reset<TPath extends Path<TForm>>(
+    pathOrStateOrUndefined?: TPath | Partial<ResetState<TForm>>,
+    stateOrOptsOrUndefined?: Partial<ResetState<TForm>> | SetValueOptions,
+    optsOrUndefined?: SetValueOptions,
+  ): Promise<void> {
+    // Extract state, path, and options based on arguments
+    // let path: TPath | undefined;
+    let state: Partial<ResetState<TForm>> | undefined;
+    let opts: SetValueOptions | undefined = optsOrUndefined;
+
+    // Case 1: reset()
+    if (!pathOrStateOrUndefined && !stateOrOptsOrUndefined) {
+      wasSubmitted.value = false;
+      submitAttemptsCount.value = 0;
+      isSubmitAttempted.value = false;
+
+      form.revertValues();
+      form.revertTouched();
+      form.revertDirty();
+
+      form.clearErrors();
+      form.clearSubmitErrors();
```

Comment:
I'm wondering if these actions can be done in all cases? If so, we can omit the repetition for each case?

Reply:
Good question, I think these would run only if they reset the whole form. If they reset a path then only that path's errors, submit errors, touched, dirty should be reset, leaving the form's own state (submitCount, etc..) unchanged.